msal/application.py

```diff
@@ -650,13 +700,27 @@ def _build_client(self, client_credential, authority, skip_regional_client=False
         if self.app_version:
             default_headers['x-app-ver'] = self.app_version
         default_body = {"client_info": 1}
-        if isinstance(client_credential, dict):
+        if ManagedIdentity.is_managed_identity(client_credential):
+            # Federated Identity Credential (FIC), a.k.a. workload identity
+            # https://learn.microsoft.com/entra/workload-id/workload-identity-federation
+            client_assertion_type = Client.CLIENT_ASSERTION_TYPE_JWT
+            client_assertion = AutoRefresher(
+                lambda: ManagedIdentityClient(
+                    client_credential, self.http_client,
+                    ).acquire_token_for_client(
+                        resource="api://AzureADTokenExchange",
```

Comment:
Does this `acquire_token_for_client` call generate an OIDC token?

What is the point of making an app trust managed identity when the managed identity can trust an external identity provider by itself?

Reply:
> Does this `acquire_token_for_client` call generate an OIDC token?

Azure CLI's "OIDC token" is a vague term that I have a hard time remembering what it really is. :-)

In the current CCA-federated-by-MI case, the underlying federation token is conceptually an access token for a specific audience.

> What is the point of making an app trust managed identity when the managed identity can be used as an identity/service principal by itself?

My understanding is that there are several advantages for sticking with CCA as your app's identity.

1. MI has operational limitations. If you deploy your popular app into a fleet of VMs, each of those VMs has a different system-assigned MI. Not sure whether it is possible to customize those MI's branding. Of course, you could configure those VMs to share one user-assigned MI, but then it hits the next limitation.
2. MI has functional limitations. It supports only service-to-service call. There is no plan to add OBO to it.
3. A CCA federated by a system-assigned or user-assigned MI gives you the best of both worlds. You have full capabilities of a CCA, yet you do not need to manage a client secret or a cert.